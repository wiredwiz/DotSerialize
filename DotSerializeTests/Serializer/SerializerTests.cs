using System;
using System.IO;
using System.Xml;
using ApprovalTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Org.Edgerunner.DotSerialize.Tests.DataTypes;

namespace Org.Edgerunner.DotSerialize.Tests
{
   /// <summary>
   /// Summary description for SerializerTests
   /// </summary>
   [TestClass]
   public class SerializerTests
   {
      protected const string _SerializeDogResultsInProperFile = "SerializerTests.SerializeDogResultsInProperFile.approved.xml";

      [TestInitialize]
      private void Setup()
      {
         Utilities.ExtractEmbeddedFile(_SerializeDogResultsInProperFile);
      }
      [TestCleanup]
      private void CleanUp()
      {
         Utilities.DeleteFile(_SerializeDogResultsInProperFile);
      }

      [TestMethod]
      public void SerializeDogResultsInProperFile()
      {
         var dog = new Dog("Fido", "Golden Retriever", true, null);
         var serializer = new Serializer();
         string xml = serializer.SerializeObject(dog);
         Approvals.VerifyXml(xml);
      }
   }
}
