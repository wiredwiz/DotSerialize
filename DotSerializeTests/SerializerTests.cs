using System;
using System.IO;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Org.Edgerunner.DotSerialize.Tests.DataTypes;

namespace Org.Edgerunner.DotSerialize.Tests
{
   /// <summary>
   /// Summary description for SerializerTests
   /// </summary>
   [TestClass]
   public class SerializerTests
   {
      [TestInitialize]
      private void Setup()
      {
         Utilities.ExtractEmbeddedFile("foo");
      }
      [TestCleanup]
      private void CleanUp()
      {
         File.Delete("foo");
      }

      [TestMethod]
      public void SerializeDogResultsInProperFile()
      {
         var dog = new Dog("Fido", "Golden Retriever", true, null);
         var serializer = new Serializer();
         serializer.SerializeObjectToFile<Dog>("C:\\DogSerializeTest.xml", dog);
         
      }
   }
}
