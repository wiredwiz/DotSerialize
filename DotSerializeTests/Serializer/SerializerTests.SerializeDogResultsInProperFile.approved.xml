<Dog xmlns:dts="http://www.dotserialize.com/Schema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.Dog,  DotSerializeTests" dts:id="1" dts:src="true">
  <Name>Fido</Name>
  <Breed>Golden Retriever</Breed>
  <BirthDate>01/01/0001 00:00:00</BirthDate>
  <_Age>0</_Age>
  <_Owner xsi:nil="true" />
  <HasCollar>True</HasCollar>
  <Nose>Normal</Nose>
  <Bone dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.Bone,  DotSerializeTests" Type="Chicken" Width="0" Diameter="0" />
  <Collar xsi:nil="true" />
</Dog>